pkg/dockerregistry/server/auth.go

```diff
@@ -248,12 +250,18 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				if verifiedPrune {
 					continue
 				}
+				if len(bearerToken) == 0 {
+					return nil, ac.wrapErr(ctx, ErrTokenRequired)
+				}
 				if err := verifyPruneAccess(ctx, osClient); err != nil {
 					return nil, ac.wrapErr(ctx, err)
 				}
 				verifiedPrune = true
 			default:
 				if err := verifyImageStreamAccess(ctx, imageStreamNS, imageStreamName, verb, osClient); err != nil {
+					if len(bearerToken) == 0 && err == ErrOpenShiftAccessDenied {
+						return nil, ac.wrapErr(ctx, ErrTokenRequired)
+					}
```

Comment:
why is prune handled different from default?  Just because you know prune will never be allowed without a valid token?

Reply:
for pruning, we check the permissions to delete images (verifyPruneAccess) because `/admin/blobs` is for deleting the blobs itself and not its layer links.

The `default` case is for all other requests that must be made within repository (imagestream).

Reply:
my question is:  why don't you invoke "verifyPruneAccess" with whatever token you have (and allow it to be rejected if the token is empty), just like you do for "verifyImageStreamAccess" ?